arithmetization/src/main/java/net/consensys/linea/rpc/linea/LineaSetExtraData.java

```diff
@@ -0,0 +1,124 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package net.consensys.linea.rpc.linea;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.extern.slf4j.Slf4j;
+import net.consensys.linea.extradata.LineaExtraDataException;
+import net.consensys.linea.extradata.LineaExtraDataHandler;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.hyperledger.besu.datatypes.rpc.JsonRpcResponseType;
+import org.hyperledger.besu.datatypes.rpc.RpcMethodError;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.parameters.JsonRpcParameter;
+import org.hyperledger.besu.plugin.services.RpcEndpointService;
+import org.hyperledger.besu.plugin.services.exception.PluginRpcEndpointException;
+import org.hyperledger.besu.plugin.services.rpc.PluginRpcRequest;
+
+@Slf4j
+public class LineaSetExtraData {
+
+  private static final AtomicInteger LOG_SEQUENCE = new AtomicInteger();
+  private final JsonRpcParameter parameterParser = new JsonRpcParameter();
+  private final RpcEndpointService rpcEndpointService;
+  private LineaExtraDataHandler extraDataHandler;
+
+  public LineaSetExtraData(final RpcEndpointService rpcEndpointService) {
+    this.rpcEndpointService = rpcEndpointService;
+  }
+
+  public void init(final LineaExtraDataHandler extraDataHandler) {
+    this.extraDataHandler = extraDataHandler;
+  }
+
+  public String getNamespace() {
+    return "linea";
+  }
+
+  public String getName() {
+    return "setExtraData";
+  }
+
+  public Boolean execute(final PluginRpcRequest request) {
+    if (log.isDebugEnabled()) {
+      // no matter if it overflows, since it is only used to correlate logs for this request,
+      // so we only print callParameters once at the beginning, and we can reference them using the
+      // sequence.
+      LOG_SEQUENCE.incrementAndGet();
+    }
+
+    final var extraData = parseRequest(request.getParams());
+
+    try {
+      updatePricingConf(extraData);
+
+      updateStandardExtraData(extraData);
+
+      return Boolean.TRUE;
+    } catch (final LineaExtraDataException lede) {
+      throw new PluginRpcEndpointException(new ExtraDataPricingError(lede));
+    }
+  }
+
+  private void updateStandardExtraData(final Bytes32 extraData) {
+    final var resp =
+        rpcEndpointService.call("miner_setExtraData", new Object[] {extraData.toHexString()});
+    if (!resp.getType().equals(JsonRpcResponseType.SUCCESS)) {
+      throw new LineaExtraDataException(
+          LineaExtraDataException.ErrorType.FAILED_CALLING_SET_EXTRA_DATA,
+          "miner_setExtraData failed: " + resp);
```

Comment:
miner_setExtraData or linea_setExtraData ? not sure what is the correct message to display to the user

Reply:
I see it could be confusion to the end user, since it makes sense mostly for us, probably better to change it to something like _internal setExtraData failed_